picotest/src/lib.rs

```diff
@@ -1,5 +1,45 @@
-pub use picotest_helpers::{run_cluster, Cluster};
+pub use std::{sync::OnceLock, time::Duration};
+
+pub use picotest_helpers::Cluster;
 pub use picotest_macros::*;
 pub mod internal;
-pub use ctor::*;
 pub use rstest::*;
+pub use std::panic;
+
+pub static SESSION_CLUSTER: OnceLock<Cluster> = OnceLock::new();
+pub static MODULE_CLUSTER: OnceLock<Cluster> = OnceLock::new();
+pub static SESSION_CLUSTER_HOLDER: OnceLock<&Cluster> = OnceLock::new();
+
+#[fixture]
+pub fn cluster(#[default(".")] plugin_path: &str, #[default(5)] timeout: u64) -> &'static Cluster {
+    SESSION_CLUSTER_HOLDER.get_or_init(|| {
+        MODULE_CLUSTER.get().map(|cls| cls.stop());
+        SESSION_CLUSTER.get_or_init(|| {
+            let timeout = Duration::from_secs(timeout);
+            Cluster::new(plugin_path, timeout)
+                .expect("Failed to parse topology")
+                .run()
+                .expect("Failed to start the cluster")
+        })
+    })
+}
+
+pub fn stop_cluster() {
+    SESSION_CLUSTER.get().map(|cls| cls.stop());
+    unsafe {
+        take(&SESSION_CLUSTER_HOLDER);
+        take(&MODULE_CLUSTER);
+    }
+}
+
+unsafe fn take<T>(holder: &OnceLock<T>) -> Option<T> {
```

Comment:
я правильно, понимаю, что цель take - вызывать деструктор у содержимого лока? если да, то может быть
лучше drop_lock?
```suggestion
unsafe fn drop_lock<T>(holder: &OnceLock<T>) {
```